File: Python/DailyLeetCode/addSpaces.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        abc= set(spaces)
        
        ans= ""

        for i in range(len(s)):
            if i in abc:
                ans+= " "
            
            ans+= s[i]
        
        return ans

File: Java/DailyLeetCode/addSpaces.java
import java.util.*;
class Solution {
    public String addSpaces(String s, int[] spaces) {
        HashSet<Integer> abc= new HashSet<>();

        for(int i: spaces){
            abc.add(i);
        }

        String[] temp= s.split("");
        StringBuilder ans= new StringBuilder();

        for(int i=0; i<temp.length; i++){
            if(abc.contains(i)){
                ans.append(' ');
            }
            ans.append(temp[i]);
        }

        return ans.toString();
    }
}